src/compiler/utilitiesPublic.ts

```diff
@@ -787,7 +792,12 @@ namespace ts {
         return find(getJSDocTags(node), predicate);
     }
 
-    /** Gets all JSDoc tags of a specified kind, or undefined if not present. */
+    /** Gets all JSDoc tags that match a specified predicate */
+    export function getAllJSDocTags<T extends JSDocTag>(node: Node, predicate: (tag: JSDocTag) => tag is T): readonly T[] {
+        return getJSDocTags(node).filter(predicate);
+    }
+
+    /** Gets all JSDoc tags of a specified kind */

```

Comment:
I wish we could rid of getAllJSDocTagsOfKind now &mdash; we don't actually use it, but it's part of the public API.

Reply:
I removed the `or undefined` part since it returns an empty array never `undefined`. Should I mark `getAllJSDocTags` as internal so as not to expand the public API ?

Reply:
No, getAllJSDocTags is better, I think people will be able to figure out which to use based on the type.